### DISCUSSION TITLE
multi select and application state

### DISCUSSION THREAD

**Discussion by bastiaanterhorst** [Budibase/budibase#4048] [Help] [answered]
Hi!
Really enjoying my first steps with Budibase. One thing I am trying to do is persist state that is driven by a multi select picker.
I would like to do the following: 

1. use a multi select to allow the user to select one or more options from a list
2. as the user clicks 'set', I store the current value of the multi select in app state (with the persistent toggle)
3. I use the state to drive things in the app
4. when I reload the page, the state still works, but I'd also like to ensure that what is selected in the multi select reflects the actual current state

I can't get 4 to work. 

- simply binding to `{{ State.show_owner_ids }}` (the state variable) does not work: `budibase-client.js?v=1.0.26:1 Uncaught (in promise) TypeError: e.forEach is not a function`
- I've tried writing JS to return the values myself, by checking if state is set, but I also cannot get that to work. I've tried returning an array of ids and a string with comma separated ids. I get the same error as when I use the normal binding.

Is this possible?

Cheers,
Bastiaan

**Comment by mjashanks** [comment, marked as answer]
Thanks Bastiaan. This is bug. I have logged it here: https://github.com/Budibase/budibase/issues/4067

Thank you for reporting this. I could not find a workaround.

FYI My attempted workaround was to set a default value on save, when a value was not set, but I found a similar bug in that as well - https://github.com/Budibase/budibase/issues/4068

**Reply by bastiaanterhorst** [reply]
Thanks for looking into it!
